### ISSUE TITLE
NSVerify and ConvDual disagreement - likely ConvDual unsoundness

### ISSUE THREAD

**Issue by castrong** [sisl/NeuralVerification.jl#144]
**The following code: **
```
# Setup problem
network = NeuralVerification.read_nnet("./examples/networks/issue_net.nnet")
input_set = NeuralVerification.Hyperrectangle([-0.0062371253401238835, 0.18329439564391525, 0.46157411780388147, -0.3298488323352671], [0.7984413176613196, 0.7984413176613196, 0.7984413176613196, 0.7984413176613196])
output_set_halfspace = LazySets.HalfSpace([-0.09291863543681655, -0.2176689899601838, -0.07453829739226348, 0.048948632014371496], 0.1911363393469332)
output_set_polytope = NeuralVerification.HPolytope([output_set_halfspace])
problem_halfspace = NeuralVerification.Problem(network, input_set, output_set_halfspace)
problem_polytope = NeuralVerification.Problem(network, input_set, output_set_polytope)

# Solve on NSVerify, ConvDual
println("NSVerify Result: ", NeuralVerification.solve(NeuralVerification.NSVerify(), problem_halfspace).status)
println("ConvDual Result: ", NeuralVerification.solve(NeuralVerification.ConvDual(), problem_polytope).status)
```

**With Network: **
```
//Neural Network File Format by Kyle Julian, Stanford 2016.
2, 4, 4, 8,
4, 8, 4,
This line extraneous
-6.55e4,-6.55e4,-6.55e4,-6.55e4,
6.55e4,6.55e4,6.55e4,6.55e4,
0.0,0.0,0.0,0.0,0.0,
1.0,1.0,1.0,1.0,1.0,
0.6140225497380456, -0.10164818650426977, -0.8124684411237504, -0.1680253578199915,
-0.4574323429591405, 0.6938035333549069, -0.7569449241683932, -0.8472273173184637,
0.05258184755815165, 0.830217275171865, 0.19491208045742603, 0.1308144272241467,
0.911846700900834, 0.8052442602716257, 0.6783139404857033, -0.6555859415206315,
-0.2188661930511926, 0.020610939479837054, 0.6542454778420015, -0.06284759675644791,
0.2883280185473205, 0.060294030640813734, -0.9907701665023865, 0.9023433936206748,
-0.9301427129036979, 0.5594703743144818, -0.9096257124633249, 0.5711331116116605,
0.23313398762470117, 0.8655288690931147, -0.4463881835976511, 0.12287110813181723,
-0.3172197948007329,
-0.9297178066215235,
-0.5895161750242113,
-0.8435685968207425,
-0.38462859390822723,
-0.741264820673281,
0.42520352455170984,
-0.2750585944772128,
0.1711673700159242, 0.2588384573298428, 0.42197783430474445, 0.812111342759358, -0.5957385551271965, 0.08112779503760947, 0.23066821013068894, -0.6281162894811376,
0.7236871423094438, 0.8552279449648816, 0.13554631440777998, -0.2742259192622214, 0.17450438876530638, -0.9966452417855192, 0.42397431220709336, 0.00033399620438379074,
0.8974265385318683, 0.03235651045375754, -0.3393771641808896, 0.003773104518737913, -0.8362279717103203, -0.7355770018521648, -0.9266104481632982, -0.6686804303888754,
-0.07644871884821391, -0.9919956283881985, 0.6402904729438594, 0.6375570334564857, -0.09156989874286348, 0.5839081597164348, -0.6228168546830544, 0.7116678920140127,
-0.3426829930479869,
-0.3995999808293158,
0.8212246325937569,
0.7411904902618471,
```

**Leads to output: **

```
NSVerify Result: violated
ConvDual Result: holds
```
Since NSVerify is complete and ConvDual is incomplete, this represents an error. Either NSVerify is incompletely identifying a :holds as a :violated or ConvDual is unsoundly identifying a :violated as a :holds. **It's likely that ConvDual is at fault since it also disagrees with MIPVerify, Reluplex, and Planet as well.**

**Comment by changliuliu** [comment]
This one is due to floating point error: the counter example found is on the boundary of the half space constraint. And when we check the set inclusion of the output from the counter example, it returns conflicting result:

> julia> LazySets.in(output, output_set_polytope)
> true
> 
> julia> LazySets.in(output, output_set_halfspace)
> false

We may create a warning to alert the user about the floating point error.

**Comment by mforets** [comment]
this looks like an upstream bug.. but i can't reproduce it, how do i get `output` in the conflicting result above?

```julia
X_NS = solve(NSVerify(), problem_halfspace)

@show X_NS.counter_example ∈ output_set_halfspace
@show X_NS.counter_example ∈ output_set_polytope

X_NS.counter_example ∈ output_set_halfspace = true
X_NS.counter_example ∈ output_set_polytope = true
```

**Comment by changliuliu** [comment]
@mforets you need to compute the output:

```
julia> x = NeuralVerification.solve(NeuralVerification.NSVerify(), problem_halfspace).counter_example
4-element Array{Float64,1}:
 0.7922041923211958
 0.9817357133052349
 0.3398047369186449
 0.12519412643245326

julia> output = NeuralVerification.compute_output(network, x)
4-element Array{Float64,1}:
  0.07768723948819561
 -0.5762273280928935
  0.28897399484750297
  1.9299362784322858

julia> LazySets.in(output, output_set_polytope)
true

julia> LazySets.in(output, output_set_halfspace)
false
```

**Comment by mforets** [comment]
ok,, thanks ! the bug should be fixed in the latest release, [v1.37.6](https://github.com/JuliaReach/LazySets.jl/releases/tag/v1.37.6).
